ruleset/sca/sunos/cis_solaris11.4.yml

```diff
@@ -700,42 +689,42 @@ checks:
     remediation: "Perform the following to implement the recommended state: Edit /etc/default/passwd - MAXWEEKS=13 MINWEEKS=1 WARNWEEKS=1"
     compliance:
       - cis: ["7.1"]
     condition: all
     rules:
-      - 'c:grep WEEKS= /etc/default/passwd -> r:^MAXWEEKS=13'
-      - 'c:grep WEEKS= /etc/default/passwd -> r:^MINWEEKS=1'
-      - 'c:grep WEEKS= /etc/default/passwd -> r:^WARNWEEKS=1'
+      - 'f:/etc/default/passwd -> !r:^# && r:MAXWEEKS && n:=\s*\t*(\d+) compare <= 13'
+      - 'f:/etc/default/passwd -> !r:^# && r:MINWEEKS && n:=\s*\t*(\d+) compare == 1'
+      - 'f:/etc/default/passwd -> !r:^# && r:WARNWEEKS && n:=\s*\t*(\d+) compare == 4'
 
   - id: 9055
-    title: "Set Strong Password Creation Policies (Scored)"
+    title: "Set Strong Password Creation Policies."
     description: "The variables in the /etc/default/passwd file indicate various strategies for creating differences required between an old and a new password. As requiring users to select a specific numbers of differences between the characters in the existing password and the new one can strengthen the password by increasing the symbol-set space, to further increase the difficulty of breaking any password by brute-force attacks, these values should be set as appropriate to the needs of the user."
     rationale: "Administrators may wish to add site-specific dictionaries to the DICTIONLIST parameter."
     remediation: "Edit /etc/default/passwd with these values: PASSLENGTH=14 NAMECHECK=YES HISTORY=10 MINDIFF=3 MINUPPER=1 MINLOWER=1 MINSPECIAL=1 MINDIGIT=1 MAXREPEATS=1 WHITESPACE=YES DICTIONDBDIR=/var/passwd DICTIONLIST=/usr/share/lib/dict/words"
     compliance:
       - cis: ["7.2"]
     condition: all
     rules:
-      - 'c:grep ^PASSLENGTH= /etc/default/passwd -> r:14'
-      - 'c:grep NAMECHECK= /etc/default/passwd -> r:YES'
-      - 'c:grep ^HISTORY= /etc/default/passwd -> r:10'
-      - 'c:grep MINDIFF= /etc/default/passwd -> r:3'
-      - 'c:grep ^MINUPPER= /etc/default/passwd -> r:1'
-      - 'c:grep ^MINLOWER= /etc/default/passwd -> r:1'
-      - 'c:grep ^MINSPECIAL= /etc/default/passwd -> r:1'
-      - 'c:grep ^MINDIGIT= /etc/default/passwd -> r:1'
-      - 'c:grep ^MAXREPEATS= /etc/default/passwd -> r:1'
-      - 'c:grep WHITESPACE= /etc/default/passwd -> r:YES'
-      - 'c:grep DICTIONDBDIR= /etc/default/passwd -> r:/var/passwd'
-      - 'c:grep ^DICTIONLIST= /etc/default/passwd -> r:/usr/share/lib/dict/words'
+      - 'f:/etc/default/passwd -> !r:^# && r:PASSLENGTH && n:=\s*\t*(\d) compare == 14'
```

Comment:
is the ! before the r correct?

Reply:
Yes, it is to ensure the line is not commented